core/sql/optimizer/NAFileSet.cpp

```diff
@@ -161,6 +162,18 @@ Lng32 NAFileSet::getKeyLength()
 	return keyLength_;
 }
 
+// returns the length of the encoded key in bytes for this index
+Lng32 NAFileSet::getEncodedKeyLength()
+{
+	if(encodedKeyLength_ >0) return encodedKeyLength_;
+
+	for(CollIndex i=0;i<indexKeyColumns_.entries();i++)
+	{
+		encodedKeyLength_ += indexKeyColumns_[i]->getType()->getNominalSize();
```

Comment:
What I have found is that the encoded keys for a salted table do not
contain length field for VARCHAR key column. Hence this new method.

Let me double check.

On Tue, Oct 20, 2015 at 11:07 AM, Hans Zeller <notifications@github.com>
wrote:

> In core/sql/optimizer/NAFileSet.cpp
> <https://github.com/apache/incubator-trafodion/pull/132#discussion_r42516287>
> :
>
> > @@ -161,6 +162,18 @@ Lng32 NAFileSet::getKeyLength()
> >  	return keyLength_;
> >  }
> >
> > +// returns the length of the encoded key in bytes for this index
> > +Lng32 NAFileSet::getEncodedKeyLength()
> > +{
> > +	if(encodedKeyLength_ >0) return encodedKeyLength_;
> > +
> > +	for(CollIndex i=0;i<indexKeyColumns_.entries();i++)
> > +	{
> > +		encodedKeyLength_ += indexKeyColumns_[i]->getType()->getNominalSize();
>
> This should also add the null indicator length for nullable types.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/132/files#r42516287>.
>



-- 
Regards, --Qifan